osc-dm-product-srv/src/utl/tag_generator.py

```diff
@@ -0,0 +1,28 @@
+from utl.openaillm import OpenAIClient
+import yaml
+import logging
+
+logger = logging.getLogger(__name__)
+def _create_tags(data, available_tags, vendor=None, model="gpt-4o",):
```

Comment:
you use gpt-3.5 earlier in this PR. Unless there is a good reason to use different models (and if there is leave a comment explaining) try to stick to just one. Ideally as a configurable value you stick in a conf value so the user of the server can change it easily.